server/src/main/java/org/opensearch/index/mapper/DerivedField.java

```diff
@@ -18,17 +18,21 @@
 import org.opensearch.script.Script;
 
 import java.io.IOException;
+import java.util.Map;
 import java.util.Objects;
 
 /**
  * DerivedField representation: expects a name, type and script.
  */
 @PublicApi(since = "2.14.0")
 public class DerivedField implements Writeable, ToXContentFragment {
-
     private final String name;
     private final String type;
     private final Script script;
+    private String sourceIndexedField;
+    private Map<String, String> properties;
+    private Boolean ignoreMalformed;
+    private String format;
```

Comment:
That could work.

From what I can tell, right now you're implementing the approach you described in https://github.com/opensearch-project/OpenSearch/issues/13143#issuecomment-2050277893

Down the road, if we decide to support the model that @reta described in https://github.com/opensearch-project/OpenSearch/issues/13143#issuecomment-2050387233, we could probably tell the difference based on the type of the value for the property. If it's a string, we use the logic you've put here, but if it's an object/map (since they're the same thing in JSON), we could do the "fancier" logic.

I've been trying to figure out where that "fancy" logic lives right now. So far as I can tell, it's handled by `ObjectMapper.TypeParser.parseProperties()`. You give it an `ObjectMapper.Builder`, and it embeds the nested `Mapper` instances for the object fields. How do we hook that up? I'm not sure. Maybe we'll need a `DerivedObjectFieldMapper`?

Reply:
You're right. 
I have defined a parser for properties in next PR, I will take care of this requirement in that PR itself - https://github.com/opensearch-project/OpenSearch/pull/13720/files#diff-14a9f3c687532d5bec9cf3d510a66dce6cfe38026b07eaae85fc2be2e73a96c5R81
For now, we can just accept the type.

Reply:
I have changed the subsequent PR to accept the object too. For now it will thrown an exception for anything other than String value, but APIs are compatible to support `properties` value as `Object`.